### PR TITLE
Move metadata settings into its own object.

### PR DESCRIPTION
**Pull request by chriseth**
@marenz please check - I think it should be one level deeper actually.

Follow-up of https://github.com/ethereum/solidity/pull/6534/

### CODE REVIEW

**Comment by Marenz** [comment]
LGTM. Looking at the documentation again, it seems i completely missed the "metadata" layer between "settings" and "useLiteralContent"

**Review by Marenz — DISMISSED** [review]

**Review by axic — CHANGES_REQUESTED** [review]
Need to update the documentation in this same PR.

Otherwise looks good to me.

File: test/libsolidity/Metadata.cpp [hunk 1]
```diff
@@ -186,22 +186,29 @@ BOOST_AUTO_TEST_CASE(metadata_useLiteralContent)
...
-		BOOST_CHECK(metadata["settings"].isMember("useLiteralContent"));
-		BOOST_CHECK(_check == metadata["settings"]["useLiteralContent"].asBool());
 		BOOST_CHECK(dev::test::isValidMetadata(metadata_str));
-
+		BOOST_CHECK(metadata.isMember("settings"));
+		if (_literal)
+		{
+			BOOST_CHECK(metadata["settings"].isMember("metadata"));
+			BOOST_CHECK(metadata["settings"]["metadata"].isMember("useLiteralContent"));
+			BOOST_CHECK(metadata["settings"]["metadata"]["useLiteralContent"].asBool());
```
Comment (axic) [inline test/libsolidity/Metadata.cpp]: It should also check here that the value is `true`.

**Comment by chriseth** [comment]
Updated.

**Review by Marenz — APPROVED** [review]

**Review by jonesk7734 — COMMENTED** [review]
😃

**Pull request merged**